src/api/useTerminologyServiceApi.ts

```diff
@@ -48,6 +48,27 @@ export class TerminologyServiceApi {
       });
     return "failure";
   }
+
+  async logoutUMLS(): Promise<Boolean> {
+    const baseUrl = await getServiceUrl();
+    const resp = await axios
+      .delete(`${baseUrl}/vsac/umls-credentials`, {
+        headers: {
+          Authorization: `Bearer ${this.getAccessToken()}`,
+          "Content-Type": "text/plain",
+        },
+        timeout: 15000,
+      })
+      .then((resp) => {
+        if (resp.status === 200) {
```

Comment:
async logoutUMLS(): Promise<boolean> {
  try {
    const baseUrl = await getServiceUrl();
    const resp = await axios.delete(`${baseUrl}/vsac/umls-credentials`, {
      headers: {
        Authorization: Bearer ${this.getAccessToken()},
        "Content-Type": "text/plain",
      },
      timeout: 15000,
    });

    // Check response status and return true if successful
    return resp.status === 200;
  } catch (error) {
    // Log the error or handle it as needed
    console.error("Logout failed:", error);
    return false;
  }
}

--> I think you can use the above optimized code which removes the unnecessary .then .catch
--> this also ensures proper error handling and removes redundant return statements

Reply:
@sb-prateekkeerthi I added error log, but i think .then and .catch are ok, same thing as try...catch